js-exercises/minima/minima.js

```diff
@@ -0,0 +1,39 @@
+const isNumber = (num) => typeof num === 'number';
+
+function minima(numberOfElementsToReturn, array) {
+  const isValidNumber = isNumber(numberOfElementsToReturn);
+
+  if (!isValidNumber) {
+    throw new TypeError(`expected number , got ${typeof numberOfElementsToReturn}`);
+  }
+
+  if (numberOfElementsToReturn < 0) {
+    throw new Error('length must be greater or equal to zero.');
+  }
+
+  if (!Array.isArray(array)) {
+    throw new TypeError(`expected array , got ${typeof array}`);
+  }
+
+  const inputArray = array.filter(element => isNumber(element));
+  const sortedArray = inputArray.sort((firstNum, secondNum) => firstNum - secondNum);
+  const result = [];
+
+  const lengthOfInputArray = inputArray.length;
+  if (lengthOfInputArray <= numberOfElementsToReturn) {
+    return sortedArray;
+  }
+
+  let index = 0;
+  for (const num of inputArray) {
```

Comment:
You can avoid this for loop by simply using Array.protype.splice method
```suggestion
  return inputArray.splice(0, numberOfElementsToReturn);
```

Reply:
Thanks for suggesting this.

Reply:
used slice method.